plugin.rb

```diff
@@ -211,6 +211,73 @@ def limit_accepts
   TopicView.add_post_custom_fields_whitelister do |user|
     ["is_accepted_answer"]
   end
+  
+  def before_head_close_meta(controller)
+    return "" if not controller.instance_of? TopicsController
+    topic = Topic.with_deleted.where(id: controller.params[:topic_id]).first
+    return "" if not controller.guardian.allow_accepted_answers_on_category?(topic.category_id)
+    first_post = topic.first_post
+    question_json =  {
+      '@type' => 'Question',
+      'name' => topic.title,
+      'text' => first_post.raw,
+      'upvoteCount' => first_post.like_count,
+      'answerCount' => topic.reply_count,
+      'dateCreated' => first_post.created_at,
+      'author' => {
+        '@type' => 'Person',
+        'name' => first_post.user.username
```

Comment:
This is a bit risky, can you change all of this to `&.` sometimes posts have no users, we don't want it to explode